containers/R/rclipper_user/vignettes/Rclipper.Rmd

```diff
@@ -0,0 +1,133 @@
+
+* Raw Vector
+* Integer Vector
+* Numeric Vector
+* String (length-1 character vector)
+* Data Frame
+* Matrix
+* Array
+* List
+
+Additionally, **given a list of inputs** of length *N*, a prediction function **must return a list of outputs**
+of length *N*. All elements of the output list must be of the same type.
+
+**Note:** If a prediction function returns a list of string (length-1 character vector) objects, each output
+will be delivered to Clipper as-is, without any additional serialization. Otherwise, all non-string outputs
+will be string-serialized via the [jsonlite package](https://cran.r-project.org/web/packages/jsonlite/index.html),
+and their serialized representations will be sent to Clipper.
+
+## Building a model
+Once you've written an API-compatible prediction function, you can build a Clipper model with it via
+the **build_model** function:
+
+```
+#' @param model_name string (character vector of length 1). The name to assign to the model image.
+#' @param model_version string (character vector of length 1). The version tag to assign to the model image.
+#' @param prediction_function function. This should accept a type-homogeneous list of 
+#' inputs and return a list of outputs of the same length. If the elements of the output list
+#' are not strings (character vectors of length 1), they will be converted to a serialized
+#' string representation via 'jsonlite'.
+#' @param sample_input For a prediction function that accepts a list of inputs of type X,
+#' this should be a single input of type X. This is used to validate the compatability
+#' of the function with Clipper and to determine the Clipper data type (bytes, ints, strings, etc)
+#' to associate with the model.
+#' @param model_registry string (character vector of length 1). The name of the image registry
+#' to which to upload the model image. If NULL, the image will not be uploaded to a registry.
+
+Rclipper::build_model(model_name, model_version, prediction_function, sample_input, model_registry = NULL)
+```
+
+This will build a docker image with the tag: *model_registry*/*model_name*:*model_version*. If no registry
+was specified, the image will have the tag: *model_name*:*model_version*. Additonally, this function will output
+a command that you can execute within an interactive Python environment to deploy the model with the 
+[clipper_admin package](https://pypi.python.org/pypi/clipper_admin).
+
+## Deploying a model
+Once you've built a model, use the provided command to deploy it with the 
+[clipper_admin package](https://pypi.python.org/pypi/clipper_admin). For information about how to
+register the model with an application so that it can be queried, please consult the 
+[clipper_admin API documentation](http://docs.clipper.ai/en/).
+
+## Querying a model
+After you've built a model, deployed the model, and registered the model with an application,
+you can query it with the input data of the correct type. The following table maps the input type of 
```

Comment:
s/with the input data of the correct type/ with input data of the correct type

(remove the "the" before "input data")